### ISSUE TITLE
[Android] FlashList is caching the wrong props for its items

### ISSUE THREAD

**Issue by alabsi91** [Shopify/flash-list#907]
## Current behavior

`FlashList` is caching the wrong props for its items. This issue occurs with lists containing around 40 items or more. In this example, the checkbox should default to 'off' based on the boolean props passed. However, for some reason, the props change every time the list updates. This issue specifically affects the item at the end of the list. This is not the first time I've encountered this problem; it happened before when passing an image URL. 
I tested it with `FlatList`, and it doesn't have this issue.

https://github.com/Shopify/flash-list/assets/58077449/a9f8bfd5-b4c3-4ca8-a4e7-e39abd26870f

## Platform:

- [x] Android paper

## Environment

@shopify/flash-list that v1.5.0

**Comment by OakMegaEdward** [comment]
We have the same issue. When the screen goes out of sight while scrolling and comes back. The avatar Image `source` back to load `defaultSource` imageUri

```
<FlashList
    data={displayList}
    keyExtractor={item => item.service_list_id}
    renderItem={({ item }) => renderFlatListItem(item)}
    onEndReachedThreshold={0.5}
    onEndReached={onScrollLoad}
    estimatedItemSize={103}
  />
```

the avatar image in `renderItem`
```
<Image
    style={styles.avatarCircle}
    onError={() => setImageError(true)}
    source={
      imageError
        ? require('@/assets/images/components/OakMegaAvatar/avatar-default.png')
        : {
            uri: imageUri
          }
    }
    defaultSource={require('@/assets/images/components/OakMegaAvatar/avatar-default.png')}
  />
```

https://github.com/Shopify/flash-list/assets/61536315/ea943e24-bf9e-4257-847f-1890a0684ea2

**Comment by huuthinh95** [comment]
I have the same issue when use FlashList (v1.6.1) with [Swipeable](https://docs.swmansion.com/react-native-gesture-handler/docs/api/components/swipeable/). When scrolling, FlashList cache wrong state of Swipeable open.
My `FlashList`:
```javascript
<FlashList
    data={data}
    renderItem={renderItem}
    keyExtractor={keyExtractor}
    estimatedItemSize={68}
    onEndReached={onLoadMore}
    onEndReachedThreshold={0.2}
 />
```
And the `renderItem`:
```javascript
const renderItem = useCallback(
    ({item, index}: {item: IConversationItem; index: number}) => {
      return (
        <Swipeable renderRightActions={renderRightActions} friction={3.5}>
           <Text>{'item ' + index}</Text>
        </Swipeable>
      );
    },
    [],
  );
```
https://github.com/Shopify/flash-list/assets/85934362/df96d23e-7276-409a-99ab-3214eb9552a7

**Comment by mohamed2m2018** [comment]
I have similar experience

**Comment by huuthinh95** [comment]
I think the problem is this: [recycling](https://shopify.github.io/flash-list/docs/recycling/). You can try and give me feedback?

**Comment by alabsi91** [comment]
The workaround is to avoid using `useState` or a component that does so in your list item, and consider that each item's rerender is essentially treated as a first render, making the use of `useEffect` pointless.

**Comment by Acetyld** [comment]
Same problem here with Swipeable

**Comment by KolissnikBogdan** [comment]
+ 1

**Comment by baesumin** [comment]
+1

**Comment by DePavlenko** [comment]
> > Adding "key={someId}" prop to the rendered item solved the issue with "Swipeable" component for us
> 
> Using key prop will lose all the benefits of using it over FlatList https://shopify.github.io/flash-list/docs/fundamentals/performant-components#remove-key-prop

Indeed. So it seems there is no solution for "Swipable" without loosing performacnce?

**Comment by alabsi91** [comment]
Has anyone tried using a class component or wrapping the item function component with a class component?

**Comment by BrunoMPedraza** [comment]
You can pass down on extraData whatever you want it to refresh at, just make sure children props are managed on FlashList container so you can put them on extraData.

**Comment by projodesign** [comment]
> > > Adding "key={someId}" prop to the rendered item solved the issue with "Swipeable" component for us
> > 
> > 
> > Using key prop will lose all the benefits of using it over FlatList https://shopify.github.io/flash-list/docs/fundamentals/performant-components#remove-key-prop
> 
> Indeed. So it seems there is no solution for "Swipable" without loosing performance?

I was also experiencing this issue with Swipeable, so added the key prop. This does solve the issue of the swiper swiping multiple items in the list, and for me the performance hit isn't too bad. It's certainly still a lot better that FlatList on a list of 300+ items.

Interestingly it only seems to affect the renderLeftActions and renderRightActions. If you are performing an action on swiping (using onSwipeableOpen for example) it does only perform that action on the item you swiped, and not all the ones that visually appear to have been swiped.
